xclim/core/indicator.py

```diff
@@ -261,26 +294,49 @@ def __call__(self, *args, **kwds):
             das[self._parameters[i]] = ba.arguments.pop(self._parameters[i])
 
         # Metadata attributes from templates
-        attrs = self.update_attrs(ba, das)
-        vname = attrs.pop("var_name")
+        var_id = None
+        var_attrs = []
+        for attrs in self.cf_attrs:
+            if n_outs > 1:
+                var_id = f"{self.identifier}.{attrs['var_name']}"
+            var_attrs.append(
+                self.update_attrs(ba, das, attrs, names=self._cf_names, var_id=var_id)
+            )
 
         # Pre-computation validation checks on DataArray arguments
         self.bind_call(self.datacheck, **das)
         self.bind_call(self.cfcheck, **das)
 
         # Compute the indicator values, ignoring NaNs and missing values.
-        out = self.compute(**das, **ba.kwargs)
+        outs = self.compute(**das, **ba.kwargs)
+        if isinstance(outs, tuple) and len(outs) != n_outs:
+            raise ValueError(
+                f"Indicator {self.identifier} was wrongly defined. Expected {n_outs} outputs, got {len(outs)}."
+            )
+        if n_outs == 1:
+            outs = [outs]
 
         # Convert to output units
-        out = convert_units_to(out, self.units, self.context)
+        outs = [
+            convert_units_to(out, attrs.get("units", ""), self.context)
+            for out, attrs in zip(outs, var_attrs)
+        ]
 
-        # Update netCDF attributes
-        out.attrs.update(attrs)
+        # Update variable attributes
+        for out, attrs in zip(outs, var_attrs):
+            var_name = attrs.pop("var_name")
+            out.attrs.update(attrs)
+            out.name = var_name
 
         # Mask results that do not meet criteria defined by the `missing` method.
+        # This means all variables must have the same dimensions...
         mask = self.mask(*das.values(), **ba.arguments)
+        outs = [out.where(~mask) for out in outs]
 
-        return out.where(~mask).rename(vname)
+        # Return a single DataArray in case of single output, otherwise a tuple
+        if n_outs == 1:
+            return outs[0]
+        return tuple(outs)
```

Comment:
Shouldn't `outs` already be a tuple ?

Reply:
Because I use list-comprehension, it's a list. But multiple outputs in python is conventionally a tuple.